subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/instantexecution/InstantExecutionProblemsFixture.groovy

```diff
@@ -305,21 +306,8 @@ final class InstantExecutionProblemsFixture {
     }
 
     private static Map<String, Object> readJsModelFrom(File reportFile) {
-        // InstantExecutionReport ensures the pure json model can be read
-        // by looking for begin-report-data and end-report-data
-        def jsonText = ""
-        def beginFound = false
-        def endFound = false
-        reportFile.eachLine { line ->
-            if (line.contains("begin-report-data")) {
-                beginFound = true
-            } else if (line.contains("end-report-data")) {
-                endFound = true
-            } else if (beginFound && !endFound) {
-                jsonText += "$line\n"
-            }
-        }
-        assert beginFound && endFound: "malformed report file"
+        String jsonText = substringBetween(reportFile.text, "// begin-report-data", "// end-report-data")
```

Comment:
This will load large strings for each whole report files in memory creating big chunks of garbage. The report file is ~2MB minimum (mostly the kotlin stdlib) + the data. Data can be as big as several MBs on small Android builds with "old" AGP versions for example.

I'll keep the streaming implementation but apply your other suggestions.

BTW, in https://github.com/gradle/gradle/commit/2a8e1cc5169f15528e9b8d347a12372d34ad920c I experimented with using kotlin/multiplatform instead of the deprecated kotlin/js. It cuts the minium report file size in half ootb thanks to dead code elimination and minification. We can't use it yet though as it breaks the inception/configuration-cache smoke test, but it's promising.

Reply:
Loading a few MBs in a test for simpler, smaller code doesn't sound too bad of a trade-off to me. Did it really make a difference in the execution time?